### PR TITLE
Fixes for Windows

### PR DESCRIPTION
**Pull request by Ace314159**
I changed a couple things to get this working on Windows.
1. For the same reason as #192, the gripper controller doesn't work on Windows as well, so I disabled it.
2. `M_PI` is not defined unless `_USE_MATH_DEFINES` is also defined
3. I had to export some protected functions of the joint trajectory controller to fix missing symbol errors.
4. Lastly, `cxxabi.h` doesn't exist on MSVC, so I didn't include it when MSVC was detected. It still still seems to compile fine without it.

### CODE REVIEW

**Comment by bmagyar** [comment]
Could we have `_USE_MATH_DEFINES` added in the cmake as a flag by any chance? It would move the file-based hack to a more hidden place and may avoid having to add it to more files in the future where we have a need for M_PI.

File: joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp [hunk 1]
```diff
@@ -179,39 +179,39 @@ class JointTrajectoryController : public controller_interface::ControllerInterfa
   rclcpp::Duration action_monitor_period_ = rclcpp::Duration(RCUTILS_MS_TO_NS(50));
 
   // callbacks for action_server_
-  rclcpp_action::GoalResponse goal_callback(
+  JOINT_TRAJECTORY_CONTROLLER_PUBLIC rclcpp_action::GoalResponse goal_callback(
```
Comment (bmagyar) [inline joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp:182]: Please try to make it look nice at least, the visibility macros should be out of the way of the actual code.
```suggestion
  JOINT_TRAJECTORY_CONTROLLER_PUBLIC
  rclcpp_action::GoalResponse goal_callback(
```

**Comment by bmagyar** [comment]
Also please fix uncrustify and cpplint

**Comment by Ace314159** [comment]
I could add the define using `add_definitions` and that would fix this issue as well as future issues for only the `diff_drive_controller`. This would be fine in this case because it is used in a cpp file. However, if `M_PI` were to be used in a header file, then having it in the CMake would require anyone using the header file to also have that define, which I think is not ideal.

**Comment by Karsten1987** [comment]
thanks @Ace314159 for picking this up. Let's see what the official ROS2 buildfarm is saying:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14806/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9555/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12472/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14959/)

**Comment by Ace314159** [comment]
They all seem to fail at a similar point involving logging. I don't think I modified that code, and I don't get that error. Do you have any idea what might be causing that?

**Comment by Karsten1987** [comment]
I think I misconfigured the job, let's try again:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14813/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9562/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12479/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14968/)

**Comment by Ace314159** [comment]
Looks like the only failing test is a linter one in ros2_control. I just put a fix for that.

Is the expected way to fix these linter errors to run `ament_uncrustify`. Is there a way to tell it to do it for all the packages in a workspace without having to manually specify all the packages.

**Comment by Karsten1987** [comment]
>  This would be fine in this case because it is used in a cpp file. However, if M_PI were to be used in a header file, then having it in the CMake would require anyone using the header file to also have that define, which I think is not ideal.

I agree with this logic. I think the current solution pretty neat.

**Review by Karsten1987 — APPROVED** [review]

File: diff_drive_controller/test/test_accumulator.cpp [hunk 2]
```diff
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#define _USE_MATH_DEFINES
```
Comment (destogl) [inline diff_drive_controller/test/test_accumulator.cpp:15]:
```suggestion
#ifndef _MSC_VER
#define _USE_MATH_DEFINES
#endif
```

**Review by destogl — APPROVED** [review]
It looks good now.

@Karsten1987 would it be possible to get Windows CI integrated?

**Pull request merged**